Python/pylifecycle.c

```diff
@@ -1251,17 +1251,20 @@ finalize_interp_clear(PyThreadState *tstate)
 {
     int is_main_interp = _Py_IsMainInterpreter(tstate);
 
-    /* bpo-36854: Explicitly clear the codec registry
-       and trigger a GC collection */
+    /* bpo-36854: Explicitly clear the codec registry */
```

Comment:
Does the test also pass if you also put this workaround  in "if (!is_main_interp) {"?

Reply:
Yup, they pass with no leaks if we move the 3 `Py_CLEAR` into the branch.